execution_engine_testing/tests/src/test/regression/step_regression.rs

```diff
@@ -0,0 +1,196 @@
+use std::{
+    collections::BTreeSet,
+    convert::TryInto,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use num_traits::Zero;
+use once_cell::sync::Lazy;
+
+use casper_engine_test_support::{
+    utils, InMemoryWasmTestBuilder, StepRequestBuilder, DEFAULT_ACCOUNTS,
+};
+use casper_execution_engine::{
+    core::engine_state::{
+        genesis::GenesisValidator, GenesisAccount, RewardItem, SlashItem, StepSuccess,
+    },
+    shared::transform::Transform,
+};
+use casper_types::{
+    system::auction::{
+        Bids, DelegationRate, SeigniorageRecipientsSnapshot, BLOCK_REWARD,
+        SEIGNIORAGE_RECIPIENTS_SNAPSHOT_KEY,
+    },
+    CLValue, EraId, Key, Motes, ProtocolVersion, PublicKey, SecretKey, StoredValue, U512,
+};
+
+static ACCOUNT_1_PK: Lazy<PublicKey> = Lazy::new(|| {
+    let secret_key = SecretKey::ed25519_from_bytes([200; SecretKey::ED25519_LENGTH]).unwrap();
+    PublicKey::from(&secret_key)
+});
+const ACCOUNT_1_BALANCE: u64 = 100_000_000;
+const ACCOUNT_1_BOND: u64 = 100_000_000;
+
+static ACCOUNT_2_PK: Lazy<PublicKey> = Lazy::new(|| {
+    let secret_key = SecretKey::ed25519_from_bytes([202; SecretKey::ED25519_LENGTH]).unwrap();
+    PublicKey::from(&secret_key)
+});
+const ACCOUNT_2_BALANCE: u64 = 200_000_000;
+const ACCOUNT_2_BOND: u64 = 200_000_000;
+
+fn initialize_builder() -> InMemoryWasmTestBuilder {
```

Comment:
Can we have this method somewhere in the `test/utils`? I've seen it used so many times already. It could accept a list of Genesis accounts as an argument.

Reply:
@mpapierski Could you provide some comments here, and on the other comment Mateusz has made?

Reply:
Since TestBuilder is not usable without genesis it probably makes sense to have two methods: `InMemoryTestBuilder::default()` that calls `run_genesis` with `DEFAULT_RUN_GENESIS_REQUEST`, and `InMemoryTestBuilder::new(&custom_request)`. We already have a helper function to create a genesis request from supplied accounts.

Reply:
Let's also look to action this idea outside of this PR please.  Seems worthwhile, but outside the scope of this PR.